tests/python/test_resource_descriptor.py

```diff
@@ -0,0 +1,59 @@
+'''
+Tests for in-toto attestation ResourceDescriptor protos.
+'''
+
+import unittest
+import google.protobuf.json_format as pb_json
+
+import in_toto_attestation.v1.resource_descriptor_pb2 as rdpb
+from in_toto_attestation.v1.resource_descriptor import ResourceDescriptor
+
+def create_test_desc():
+    desc = rdpb.ResourceDescriptor()
+    desc.name = 'theName'
+    desc.uri = 'https://example.com'
+    desc.digest['alg'] = 'abc123'
+    desc.content = b'bytescontent'
+    desc.downloadLocation = 'https://example.com/test.zip'
+    desc.mediaType = 'theMediaType'
+    desc.annotations['a1'].update({'keyStr': 'val1', 'keyNum': 13})
+    desc.annotations['a2'].update({'keyObj': {'subKey': 'subVal'}})
+    return ResourceDescriptor(desc)
+
+class TestResourceDescriptor(unittest.TestCase):
+    def setUp(self):
+        self.want_full_rd = '{"name":"theName","uri":"https://example.com","digest":{"alg":"abc123"},"content":"Ynl0ZXNjb250ZW50","downloadLocation":"https://example.com/test.zip","mediaType":"theMediaType","annotations":{"a1":{"keyNum": 13,"keyStr":"val1"},"a2":{"keyObj":{"subKey":"subVal"}}}}'
```

Comment:
optional: It looks like these are all only used in single tests.  My usual preference would be to define them in those tests. That way it's easier for folks reading the test to understand what's going on.  If you follow that approach you won't need setUp at all.